models/classes/LtiAgsScoreService.php

```diff
@@ -0,0 +1,79 @@
+<?php
+
+/**
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2021 (original work) Open Assessment Technologies SA (under the project TAO-PRODUCT);
+ *
+ */
+
+declare(strict_types=1);
+
+namespace oat\taoLti\models\classes;
+
+use OAT\Library\Lti1p3Ags\Factory\Score\ScoreFactory;
+use OAT\Library\Lti1p3Ags\Factory\Score\ScoreFactoryInterface;
+use OAT\Library\Lti1p3Ags\Service\Score\Client\ScoreServiceClient;
+use OAT\Library\Lti1p3Ags\Service\Score\ScoreServiceInterface;
+use OAT\Library\Lti1p3Core\Exception\LtiExceptionInterface;
+use OAT\Library\Lti1p3Core\Message\Payload\Claim\AgsClaim;
+use OAT\Library\Lti1p3Core\Registration\RegistrationInterface;
+use oat\oatbox\service\ConfigurableService;
+use Zend\ServiceManager\ServiceLocatorAwareInterface;
+use Zend\ServiceManager\ServiceLocatorAwareTrait;
+
+class LtiAgsScoreService extends ConfigurableService implements ServiceLocatorAwareInterface
+{
+    use ServiceLocatorAwareTrait;
+
+    public const SERVICE_ID = 'taoLti/LtiAgsScoreService';
+
+    public const OPTION_SCORE_SERVICE_CLIENT = 'score_service_client';
+    public const OPTION_SCORE_FACTORY = 'score_factory';
+
+    /**
+     * @throws LtiAgsException
+     */
+    public function send(RegistrationInterface $registration, AgsClaim $ags, array $data): bool
+    {
+        $scoreClient = $this->getOption(self::OPTION_SCORE_SERVICE_CLIENT) ?? new ScoreServiceClient();
+        $scoreFactory = $this->getOption(self::OPTION_SCORE_FACTORY) ?? new ScoreFactory();
+
+        if (!$scoreClient instanceof ScoreServiceInterface) {
+            throw new LtiAgsException(
+                sprintf('%s option should implement ScoreServiceInterface', self::OPTION_SCORE_SERVICE_CLIENT)
+            );
+        }
+
+        if (!$scoreFactory instanceof ScoreFactoryInterface) {
+            throw new LtiAgsException(
+                sprintf('%s option should implement ScoreFactoryInterface', self::OPTION_SCORE_FACTORY)
+            );
+        }
+
+        $score = $scoreFactory->create($data);
+
+        try {
+            return $scoreClient->publishScoreForClaim($registration, $score, $ags);
```

Comment:
I just realized we're returning a boolean here, why is that required?

Ideally, methods that produce a state change in the application should not return anything. For its clients, throwing an exception when something is wrong should be enough.